lib/response.js

```diff
@@ -368,6 +369,10 @@ res.sendFile = function sendFile(path, options, callback) {
     done = options;
     opts = {};
   }
+  
+  if(typeof opts.etag == 'undefined') {
+      opts.etag = app.set('etag');
```

Comment:
Do not pass in non-booleans to the sub module, because that is just asking for undefined behavior later. The etag option only accepts true or false.